gensim/summarization/bm25.py

```diff
@@ -225,7 +225,25 @@ def _get_scores(bm25, document):
     return bm25.get_scores(document)
 
 
-def iter_bm25_bow(corpus, n_jobs=1):
+def _bow_to_text(corpus):
+    """Helper function for transforming corpus from BOW-format to list of words format.
+
+    Parameters
+    ----------
+    corpus : corpus in BOW-format (word, weight).
+
+    Returns
+    -------
+    list of documents (list of words).
+    """
+    return [
```

Comment:
Can we avoid the [nested list comprehension]() here? I can kind of see what the code is doing after looking at it for a minute or so, but it can be so much simpler:

```python
def g():
    for doc in corpus:
        for word, _ in doc:
            yield word
return list(g())
```

This occupies nearly the same amount of lines, and would save future readers of the code much pain.

Reply:
I don't think these are equivalent: the original creates each word `count` times, right?

If so, my proposal would be a straightforward:

```python
retval = []
for doc in corpus:
    for word, count in doc:
        retval.extend([word] * count)
return retval
```

Reply:
But is `word` really a word? The documentation says `BOW-format (word, weight)`, but BOW in Gensim means tuples of `(word_id, weight)`. Not `(word string, integer count)`.

Reply:
OK, so I misread that nested list comprehension :blush:. 

To answer your other questions: I think it's all more reason to add unit tests around the code to demonstrate its behavior.

Reply:
Absolutely. The summarization module was contributed externally and I never used it, so its quality is an unknown to me. Good docs and unit tests are critical.